lib/matplotlib/tests/test_backend_pgf.py

```diff
@@ -337,3 +337,38 @@ def test_minus_signs_with_tex(fig_test, fig_ref, texsystem):
     mpl.rcParams["pgf.texsystem"] = texsystem
     fig_test.text(.5, .5, "$-1$")
     fig_ref.text(.5, .5, "$\N{MINUS SIGN}1$")
+
+
+@pytest.mark.backend("pgf")
+def test_sketch_params():
+    fig, ax = plt.subplots(figsize=[3, 3])
+    ax.set_xticks([])
+    ax.set_yticks([])
+    ax.set_frame_on(False)
+    handle = ax.plot([0, 1])[0]
+    handle.set_sketch_params(scale=5, length=30, randomness=42)
+
+    with BytesIO() as fd:
+        fig.savefig(fd, format='pgf')
+        buf = fd.getvalue().decode()
+
+    baseline = r"""\begin{pgfscope}%
+\pgfpathrectangle{\pgfqpoint{0.375000in}{0.300000in}}""" \
+    r"""{\pgfqpoint{2.325000in}{2.400000in}}%
+\pgfusepath{clip}%
+\pgfsetrectcap%
+\pgfsetroundjoin%
+\pgfsetlinewidth{1.003750pt}%
+\definecolor{currentstroke}{rgb}{0.000000,0.000000,1.000000}%
+\pgfsetstrokecolor{currentstroke}%
+\pgfsetdash{}{0pt}%
+\pgfpathmoveto{\pgfqpoint{0.375000in}{0.300000in}}%
+\pgfpathlineto{\pgfqpoint{2.700000in}{2.700000in}}%
+\pgfkeys{/pgf/decoration/.cd, """ \
+    r"""segment length = 0.300000in, amplitude = 0.050000in}%
+\pgfmathsetseed{42}%
+\pgfdecoratecurrentpath{random steps}%
+\pgfusepath{stroke}%
+\end{pgfscope}%"""
+    # check that \pgfkeys{/pgf/decoration/.cd, ...} is in path definition
```

Comment:
Maybe that's all you need in `baseline`, then?

Reply:
I stripped it down a bit, but the relative position of `\pgfdecoratecurrentpath` does matter.